src/attr/_make.py

```diff
@@ -934,9 +934,15 @@ def slots_setstate(self, state):
             Automatically created by attrs.
             """
             __bound_setattr = _obj_setattr.__get__(self)
-            for name in state_attr_names:
-                if name in state:
-                    __bound_setattr(name, state[name])
+            if isinstance(state, tuple):
```

Comment:
I think I would prefer this inverted with the default / more common case first: `if isinstance(state, dict)` or does something speak against that?

Reply:
I figured that `dict` is now the only case in `__getstate__()` so the `else` case being the default `dict` case in `__setstate__()` made sense to me here.

This is trying to handle the old `tuple` case explicitly which is known versus any other potential past unknowns (which would be the implication if inverted). Also can be changed if preferred though.

Reply:
Makes sense! I wonder if default case is faster but that doesn't seem worth it.

Reply:
👍🏻 _(It's probably moot as you'd need an `isinstance()` call either way…)_